### PR TITLE
Added high level overview of git

### LINKED ISSUE #12: Prepeare git infrastructure overview

### PR DESCRIPTION
**Pull request by graywolf**
Fixes #12

### CODE REVIEW

**Comment by graywolf** [comment]
take a look, is this at least close to what you expected? This is more bullet-point style, since I expect it will at least partically be accompanied by discussion

**Comment by voloda** [comment]
It seems to be a good starting point. I was thinking that this should also include the details on how a commit is structured (hash). etc.

File: 00-Overview/readme.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Everyone has complete history
+-----------------------------
+
+* this is very important point
+* I'm sure it happened to you at least once, you commit password into source control
+	* if someone already cloned/pulled from you, all you can do is change it
+	* committing new version without the password won't help you, because everyone has complete history, each and every commit, in the local repository
+* other cause where you have to remember this is if you commit some huge file (`.pdb` or something) and later down the line you notice  that your repository is huge
+	* deleting the file and committing the deletion won't help
```
Comment (voloda) [inline 00-Overview/readme.md]: I believe that this is a bit more complicated. My understanding:
- this is correct only if your large file stays part of a branch
- if you remove the commit your large file will disappear after retention period

Perhaps you should just note there into brackets something like (unless you rewrite the history - see below)?

**Comment by graywolf** [comment]
> It seems to be a good starting point. I was thinking that this should also include the details on how a commit is structured (hash). etc.

Good idea, will add it

**Comment by graywolf** [comment]
> this is correct only if your large file stays part of a branch
> if you remove the commit your large file will disappear after retention period

well, technically you are right, but that happens only in two cases:

1) if you delete branch containing the commit and that branch wasn't merge into anything
2) if you delete commit -> rewrite history

But I agree, I can add note like that into it

**Comment by voloda** [comment]
:+1:

**Pull request merged**